src/Microsoft.Identity.Client/AppConfig/ApplicationConfiguration.cs

```diff
@@ -12,18 +12,31 @@ namespace Microsoft.Identity.Client
 {
     internal sealed class ApplicationConfiguration : IApplicationConfiguration
     {
+        public const string DefaultClientName = "UnknownClient";
+        public const string DefaultClientVersion = "0.0.0.0";
+
         // For telemetry, the ClientName of the application.
-        public string ClientName { get; internal set; }
+        private string _clientName = DefaultClientName;
+        public string ClientName
+        {
+            get => _clientName;
+            internal set { _clientName = string.IsNullOrWhiteSpace(value) ? DefaultClientName : value; }
+        }
 
         // For telemetry, the ClientVersion of the application.
-        public string ClientVersion { get; internal set; }
+        private string _clientVersion = DefaultClientVersion;
+        public string ClientVersion
+        {
+            get => _clientVersion;
+            internal set { _clientVersion = string.IsNullOrWhiteSpace(value) ? DefaultClientVersion : value; }
+        }
 
         public bool UseCorporateNetwork { get; internal set; }
         public string IosKeychainSecurityGroup { get; internal set; }
 
         public bool IsBrokerEnabled { get; internal set; }
 
-        public IMatsConfig MatsConfig { get; internal set; }
+        public ITelemetryConfig TelemetryConfig { get; internal set; }
```

Comment:
Even if I doubt anybody used it, isn't this a breaking change?
4.0 ?

Reply:
Not the only "breaking" change in this PR. I suppose technically it requires a major version bump, however no-one can probably use this for anything today. 

---
In reply to: [285810660](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1161#discussion_r285810660) [](ancestors = 285810660)

Reply:
the problem is that this is anyway a binary breaking change, and some ADAL internal partners were very angry in the past when we made a small binary breaking change without bumping-up the major number (think of those who bin-place DLLs instead of rebuilding)
We have no choice but bump-up the major version (and I think that's fine).

Reply:
Some ADAL internal partners might want to rebuild their code instead of rebuilding :). 

The .net ecosystem does not offer any public tools for detecting these changes, so you have to rely on experienced SDEs to pick these up during PR. We are all well intention-ed here, but good intentions don't work, especially when reviewing 50+ file changes.

Reply:
no worries, @bgavrilMS : we picked it up during PR :) ?

Reply:
Well this is a proper breaking change, we would have picked this up during our release validation :)

I was talking about binary breaking changes like adding a new method to a public interface.

Reply:
adding a new method to the public interface is not breaking (even binary). And we can warn users by changing the minor version number. The CLI knows how to handle it. 
Removing or changing a public member is a breaking change (source and binary)

Reply:
Wondering if we can make this an addition and just in the migration aid show a warning for now?

Reply:
This isn't in the public API. this is ApplicationConfiguration, a class which is internal.  What we show publicly is the IAppConfig on the pca/cca which does not include the MATS infrastructure.